// openapi-to-md/dist/utils.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getApiObject = exports.getRefName = exports.convertPath = exports.tableText = exports.markdownText = exports.readDocument = void 0;
const yaml_1 = __importDefault(require("yaml"));
const readDocument = (src) => {
    try {
        const doc = yaml_1.default.parse(src);
        if (typeof doc === "object" && doc !== null)
            return doc;
    }
    catch {
        //
    }
    try {
        const doc = JSON.parse(src);
        if (typeof doc === "object" && doc !== null)
            return doc;
    }
    catch {
        //
    }
    return null;
};
exports.readDocument = readDocument;
/**
 * Replaces newlines with two spaces and a newline character, as required by
 * Markdown for a line break.
 */
const markdownText = (text) => text.replace(/\n/g, "  \n");
exports.markdownText = markdownText;
/**
 * Replaces newlines with <br> for use in Markdown tables.
 */
const tableText = (text) => text.replace(/\n/g, "<br>");
exports.tableText = tableText;
/**
 * Converts a path to a format that can be used in a URL, such as replacing
 * spaces.
 *
 * @param path Path to convert
 * @returns The converted path
 */
const convertPath = (path) => path
    .replace(/[!@#$%^&*()+|~=`[\]{};':",./<>?]/g, "")
    .replace(/ /g, "-")
    .toLowerCase();
exports.convertPath = convertPath;
/**
 * Return the name of a reference object.
 * @param refObject
 */
const getRefName = (refObject) => {
    if (typeof refObject === "object" && refObject && "$ref" in refObject) {
        return refObject["$ref"];
    }
    return undefined;
};
exports.getRefName = getRefName;
const getApiObject = ({ references }, object, refs) => {
    const refName = (0, exports.getRefName)(object);
    if (refName) {
        const ref = object["$ref"];
        if (refs) {
            if (refs.has(ref)) {
                return object;
            }
            refs.add(ref);
        }
        return references[ref];
    }
    return object;
};
exports.getApiObject = getApiObject;
